### PR TITLE
Deploy to gigalixir

### LINKED ISSUE #3: Deploy to Gigalixir
Refer: https://hexdocs.pm/phoenix/gigalixir.html#content

### PR DESCRIPTION
**Pull request by Menubatenny**
closes #3

### CODE REVIEW

File: config/prod.exs [hunk 1]
```diff
@@ -11,11 +11,13 @@ use Mix.Config
 # before starting your production server.
 config :nudge, NudgeWeb.Endpoint,
   url: [host: System.get_env("HOSTNAME", "localhost"), port: 80],
-  cache_static_manifest: "priv/static/cache_manifest.json"
+  cache_static_manifest: "priv/static/cache_manifest.json",
+  force_ssl: [rewrite_on: [:x_forwarded_proto]]
```
Comment (sreecodeslayer) [inline config/prod.exs:15]: What will this do?
Do we need it to be this?

File: .github/workflows/github-action.yml [hunk 2]
```diff
@@ -1,6 +1,7 @@
 name: Elixir CI
 
 on:
+  workflow_dispatch:
```
Comment (sreecodeslayer) [inline .github/workflows/github-action.yml:4]: hmm 🤔 
we don't want this for all jobs.

we need a new job that will help us deploy to gigalixir which should be a manual trigger. 
The test and docker builds should happen automatically

**Review by sreecodeslayer — CHANGES_REQUESTED** [review]

File: config/prod.exs (hunk 1 above)
Comment (Menubatenny) [reply config/prod.exs:15]: work flow failed in format check so i changed this
used force_ssl by refering this docs https://hexdocs.pm/phoenix/using_ssl.html#force-ssl

Comment (sreecodeslayer) [reply config/prod.exs:15]: work flow shouldn't be failing if we haven't changed on `main`. Also fix for a bad format issue is not enabling ssl. please revert this and let's fix the format issue if any

Comment (sreecodeslayer) [reply config/prod.exs:15]: btw, am asking about this commit: https://github.com/code-mancers/nudge/pull/8/commits/4a566b364e5e1dbf439b55ef66ca34a024b6cc9b
You seem to have added force ssl, any reason why this change is needed?

File: config/releases.exs [hunk 3]
```diff
@@ -36,6 +36,7 @@ config :nudge, NudgeWeb.Endpoint,
 # to start each relevant endpoint:
 #
 config :nudge, NudgeWeb.Endpoint, server: true
-#
+  http: [port: {:system, "PORT"}],
+  url: [host: System.get_env("all-friendly-flatfish") <> ".gigalixirapp.com", port: 443]
```
Comment (sreecodeslayer) [inline config/releases.exs:40]: please don't hard code this. use `HOSTNAME` env var instead

File: config/prod.exs (hunk 1 above)
Comment (Menubatenny) [reply config/prod.exs:15]: i thought we need to use ssl while deploying https://hexdocs.pm/phoenix/gigalixir.html#making-our-project-ready-for-gigalixir

Comment (Menubatenny) [reply config/prod.exs:15]: removed it

File: config/releases.exs (hunk 3 above)
Comment (sreecodeslayer) [reply config/releases.exs:41]: @Menubatenny this line is already there in the same file on L26?
either use `url` or `host` and `port` in that line, and remove this one.

File: config/prod.exs (hunk 1 above)
Comment (sreecodeslayer) [reply config/prod.exs:15]: oh yea, good catch. we can add it later on, for now its fine

File: .github/workflows/github-action.yml [hunk 4]
```diff
@@ -63,13 +63,11 @@ jobs:
...
-        ref: main
         fetch-depth: 0
-    - name: migration
-      run: gigalixir ps:migrate
     - uses: mhanberg/gigalixir-action@v0.6.0
       with:
         GIGALIXIR_USERNAME: ${{ secrets.GIGALIXIR_USERNAME }}
         GIGALIXIR_PASSWORD: ${{ secrets.GIGALIXIR_PASSWORD }}
         GIGALIXIR_APP: all-friendly-flatfish
+        MIGRATIONS: false
```
Comment (sreecodeslayer) [inline .github/workflows/github-action.yml]: why are we disabling migrations?

**Review by sreecodeslayer — APPROVED** [review]

**Pull request merged**